### PR TITLE
Component: do not render plan-storage if sitePlan is not defined

### PR DESCRIPTION
**Pull request by retrofox**
This PR fix the issue when the `sitePlan` var isn't defined. If it is the case do not render the component.

### Testing
1) Be sure to clean the browser cache
2) Go to the plan-storage testing page: `http://calypso.localhost:3000/devdocs/blocks/plan-storage`
3) Take a look the browser console. Everything should be ok.
The bug to fix is `Uncaught (in promise) TypeError: Cannot read property 'product_slug' of null`

image

### CODE REVIEW

File: client/blocks/plan-storage/index.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ class PlanStorage extends Component {
 			siteSlug,
 		} = this.props;
 
-		if ( jetpackSite ) {
+		if ( jetpackSite || ! sitePlan ) {
```
Comment (gwwar) [inline client/blocks/plan-storage/index.jsx]: Since we're not using the plan object for anything else, an alternative would be in connect to return the slug directly and check for truthyness of `sitePlanSlug` here. 
```js
const sitePlan = getSitePlan( state, siteId );
const sitePlanSlug = get( sitePlan, 'product_slug' );
```

In a future PR we should add an additional selector to cover this case

Comment (retrofox) [inline client/blocks/plan-storage/index.jsx]: do you mean create a new selector for instance?

Comment (gwwar) [inline client/blocks/plan-storage/index.jsx]: Right but we can do that in another PR

**Comment by retrofox** [comment]
> Which throws a react warning because of the missing unit value.

fixed

**Comment by retrofox** [comment]
Thanks folks for the review.
